syncopy/datatype/base_data.py

```diff
@@ -155,7 +169,71 @@ def container(self):
+        multi-threaded/parallel context, like in a compute function (cF).
+
+        Parameters
+        ----------
+            propertyName : str
+                The name for the new dataset, this will be used as the dataset name in the hdf5 container when saving.
+                It will be added as an attribute named `'_' + propertyName` to this SyncopyData object.
+                Note that this means that your propertyName must not clash with other attribute names of syncopy data objects.
+                To ensure the latter, it is recommended to use names with a prefix like `'dset_'`. Clashes will be detected and result in errors.
+            in_data : None or np.ndarray
+                The data to store. Must have the final number of dimensions you want.
+        """
+        if not propertyName in self._hdfFileDatasetProperties:
+            self._hdfFileDatasetProperties = self._hdfFileDatasetProperties + (propertyName,)
+        if inData is None:
+            setattr(self, "_" + propertyName, None)
+        else:
+            if not isinstance(inData, np.ndarray):
+                raise SPYValueError(lgl="object of type 'np.ndarray' or None", varname="inData")
+            self._set_dataset_property_with_ndarray(inData, propertyName, inData.ndim)
+
+    def _unregister_seq_dataset(self, propertyName, del_from_file=True):
+        """
+        Unregister and delete a sequential dataset from memory, and optionally delete it from the backing hdf5 file.
+
+        Assumes that the backing h5py file is open in writeable mode.
+
+        Parameters
+        ----------
+            propertyName : str
+                The name of the entry in `self._hdfFileDatasetProperties` to remove.
+                The attribute named `'_' + propertyName` of this SyncopyData object will be deleted.
+            del_from_file: bool
+                Whether to also remove the dataset named 'propertyName' from the backing hdf5 file on disk.
+        """
+        if propertyName in self._hdfFileDatasetProperties:
+            tmp_list = list(self._hdfFileDatasetProperties)
+            tmp_list.remove(propertyName)
+            self._hdfFileDatasetProperties = tuple(tmp_list)
+        if hasattr(self, "_" + propertyName):
+            delattr(self, "_" + propertyName)
+        if del_from_file:
+            if self.mode == "r":
+                lgl = "HDF5 dataset with write or copy-on-write access"
+                act = "read-only file"
+                raise SPYValueError(legal=lgl, varname="mode", actual=act)
+            if isinstance(self._data, h5py.Dataset):
+                if isinstance(self._data.file, h5py.File):
+                    if propertyName in self._data.file.keys():
+                        del self._data.file[propertyName]
+                else:
+                    SPYWarning("Could not delete dataset from file.")
+
+    def _update_seq_dataset(self, propertyName, inData=None):
```

Comment:
Here the elegance of the register/unregister interface shows :slightly_smiling_face: